developer/src/kmcmplib/src/kmx_u16.cpp

```diff
@@ -0,0 +1,257 @@
+
+//#include "../../kmcompx/include/kmcompx.h"
+#include <kmcompx.h>
+#include "kmx_u16.h"
+
+#include <xstring.h>
+#include <km_types.h>
+#include <kmx_file.h>
+#include <codecvt>
+#include <stdarg.h>
+
+//String <- wstring
+std::string string_from_wstring(std::wstring const str) {
+	std::wstring_convert<std::codecvt_utf8<wchar_t>, wchar_t> converter;
+	return converter.to_bytes(str);
+}
+//wstring <- string
+std::wstring wstring_from_string(std::string const str) {
+	std::wstring_convert<std::codecvt_utf8<wchar_t>, wchar_t> converter;
+	return converter.from_bytes(str);
+}
+
+//u16String <- string
+std::u16string u16string_from_string(std::string const str) {
+  std::wstring_convert<std::codecvt_utf8<char16_t>, char16_t> converter;
+  return converter.from_bytes(str);
+}
+
+//string <- u16string
+std::string string_from_u16string(std::u16string const str) {
+	std::wstring_convert<std::codecvt_utf8<char16_t>, char16_t> converter;
+	return converter.to_bytes(str);
+}
+
+// often used with c_str() e.g. u16fmt( DEBUGSTORE_MATCH).c_str()
+// UTF16 (= const char16_t*) -> UTF8 (= std::string)  -> UTF16 ( = std::wstring 16 bit)
+std::wstring u16fmt(const km_kbp_cp* str) {
```

Comment:
@Marc: you named this function: should we change the name?

Reply:
I don't think we change the name at this point in time. My preference is to get this PR over the line without attempting to make it too 'beautiful'... because we can do any refactoring, renaming, etc, in smaller PRs which can be easily tested and reviewed.